logos/src/logos/dbutils/dbmanager.py

```
@@ -3134,11 +3252,34 @@ def get_provider(self, provider_id: int):
         }
 
+    def update_provider_info(self, logos_key: str, provider_id: int, **kwargs) -> Tuple[dict, int]:
+        if not self.check_authorization(logos_key):
+            return {"error": "Provider changes only allowed for logos admin."}, 500
+        allowed = {"name", "base_url", "api_key", "auth_name", "auth_format",
+                   "provider_type", "cloud_provider_type", "privacy_level"}
+        updates = {k: v for k, v in kwargs.items() if k in allowed and v is not None}
+        if "provider_type" in updates:
+            updates["provider_type"] = normalize_provider_type(updates["provider_type"])
+        if "privacy_level" in updates and updates["privacy_level"] not in VALID_PRIVACY_LEVELS:
+            return {"error": f"Invalid privacy_level"}, 400
```

Flake8 (7.3.0)

[error] f-string is missing placeholders

(F541)